### PR TITLE
Enable testing with base AWS identity and optionally with assumable AWS role

### PR DESCRIPTION
**Pull request by UpGado**
- We now have the following env variables:
  * `TEST_BASE_ARN`: **automatically* determined by our root Makefile (using the aws CLI) if not set by user
  * `TEST_ASSUMABLE_ARN`: **optional**, if set by user, we run additional tests (For example, all our Python tests now are run once with base ARN and one with the assumable ARN if provided)

feedback: are these env vars names good? I thought it is better to include 'arn' in there to specify AWS-related stuff, and allow ourselves to add more clouds later without awkward naming

- Updated README.md to include working instructions (feedback on whether they are too detailed, should we automate more of them?)
- `make test` now truly works from the base directory

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
 dev:
 	make run-in-docker CMD="bash"
 dev-env: dc-build
 	$(docker_env) docker-compose up
 dc-build: ssl/rootCA.key
 	$(docker_env) docker-compose -f docker-compose.yml -f docker-compose.test.yml build
-# Runs all tests, including E2E tests
-test: run-tests-in-docker
+test: # dc-build
+	make run-in-docker CMD="make run-testsuite"
```
Comment (UpGado) [inline Makefile:15]: knit-pick

File: Makefile [hunk 2]
```diff
@@ -22,25 +27,22 @@ TEST_DBUSER=bob
...
 
-run-tests-in-docker:  dc-build  # need SSL certs for Postgres services
-	make run-in-docker CMD=""
-
-
 vault_secret = { $\
 "password": "$(TEST_DBPASS)", $\
 "iam_arns": [ $\
-	"${TEST_IAM_ROLE}" $\
+	"${TEST_ASSUMABLE_ARN}", $\
```
Comment (UpGado) [inline Makefile:39]: currently, is TEST_ASSUMABLE_ARN is not provided, there will be an empty string in the iam_arns. Is this a problem?

File: Makefile [hunk 3]
```diff
@@ -50,27 +52,18 @@ ssl/rootCA.key:
...
 	cd sdk/python && poetry run pytest --workers auto
-
-run-pg2tests: enable-vault-path seed-vault-all-hosts
-	for ADDR in $(TEST_DBADDRS); do \
-		make seed-vault-host ADDR=$$ADDR \
-		echo '###### Testing with DBADDR' $$ADDR 'SSL=ON #####'; \
-		PGSSLMODE=require PSYCOPG2_TESTDB_HOST=$$HOST make -C sdk/python/ test; \
-		echo '###### Testing with DBHOST' $$HOST 'SSL=OFF #####'; \
-		PGSSLMODE=disable PSYCOPG2_TESTDB_HOST=$$HOST make -C sdk/python/ test; \
-	done
```
Comment (UpGado) [inline Makefile:76]: if time allows, we can add this back to our test suite, but we currently don't use it, so i think it is better to deprecate it for now

File: README.md [hunk 4]
```diff
@@ -33,20 +33,26 @@ soon as we can!
...
-- To use our Python SDK to shoot a request at the authenticator, run
-  `$ PGHOST=dbmd5 PGUSER=bob PGDATABASE=db python3 sdk/python/examples/pg2_client.py`.
+- Run `$ make dc-build`. This will build the authenticator and development Docker images.
+- Run `$ docker-compose up`. This will run the authenticator with a Vault backend and will run test database servers (Postgres and MySQL).
+- In another window, `$ make dev`. This will start a shell in the development environment.
+- You now have a full development and testing environment!
+- For example, to use our Python SDK to create an Approzium connection to a Postgres server:
+    * Create an Approzium path in the test Vault backend: `$ make enable-vault-path`
+    * Give your AWS-identity access to the test server: `$ make seed-vault-addr ADDR=dbmd5:5432`
+    * Create a connection: `$ cd sdk/python/examples && poetry run python3 psycopg2_connect.py`.
```
Comment (tyrannosaurus-becks) [inline README.md:48]: ❤️ ❤️ ❤️ !!!!!!!!!!

File: authenticator/server/authenticator_test.go [hunk 5]
```diff
@@ -387,6 +387,9 @@ func TestFuzzXorBytes(t *testing.T) {
 }
 
 func TestMetrics(t *testing.T) {
+	if !testtools.ShouldRunAcceptanceTests() {
+		t.Skip("Skipping because acceptance tests are off")
+	}
```
Comment (tyrannosaurus-becks) [inline authenticator/server/authenticator_test.go:392]: Thank you and sorry I keep adding flaky tests! 😊

File: Makefile (hunk 2 above)
Comment (tyrannosaurus-becks) [reply Makefile:39]: I think that will be fine. It just looks for a match.

**Pull request merged**